waspc/src/Wasp/Generator/WebAppGenerator/ExternalCodeGenerator.hs

```diff
@@ -31,8 +33,21 @@ mkExtCodeGeneratorStrategy :: Path' (Rel C.WebAppSrcDir) (Dir GeneratedExternalC
 mkExtCodeGeneratorStrategy extCodeDirInWebAppSrcDir =
   ExternalCodeGeneratorStrategy
     { _resolveJsFileWaspImports = resolveJsFileWaspImportsForExtCodeDir (SP.castRel extCodeDirInWebAppSrcDir),
-      _extCodeDirInProjectRootDir =
-        C.webAppRootDirInProjectRootDir
-          </> C.webAppSrcDirInWebAppRootDir
-          </> extCodeDirInWebAppSrcDir
+      _resolveDstFilePath = resolveDstFilePath
     }
+  where
+    resolveDstFilePath filePath =
+      if isInStaticAssetsDir filePath
+        then
+          C.webAppRootDirInProjectRootDir
+            </> C.staticAssetsDirInWebAppDir
+            </> removeStaticAssetsDirPrefix filePath
+        else
+          C.webAppRootDirInProjectRootDir
+            </> C.webAppSrcDirInWebAppRootDir
+            </> extCodeDirInWebAppSrcDir
+            </> filePath
+
+    isInStaticAssetsDir filePath = staticAssetsDir `isPrefixOf` SP.toFilePath filePath
+    removeStaticAssetsDirPrefix filePath = fromJust $ SP.parseRelFile $ drop (length staticAssetsDir) $ SP.toFilePath filePath
```

Comment:
Here also probably better to use more precise fuction than SP.toFilePath.

Reply:
So this works? Interesting. Beacuse I would expect that a "/" remains at front of the path and then parsing it as rel file fails because it looks like an absolute file. Maybe your `staticAssetsDir` does contain "/" at the end? Ok, I am guessing from this that StrongPath's toFilePath actually adds "/" at the end of path when path is dir! That saves the day here then, and above, but still, this is something that it is very hard to rely on, and shouldn't be relied on actually.